test/NuGet.Core.Tests/NuGet.Packaging.Test/NoRefOrLibFolderInPackageRuleTests.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+using NuGet.Packaging.Rules;
+using Moq;
+using System.Xml.Linq;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+
+namespace NuGet.Packaging.Test
+{
+    public class NoRefOrLibFolderInPackageRuleTests
+    {
+        [Theory]
+        [MemberData("WarningRaisedData", MemberType = typeof(FileSource))]
+        public void WarningRaisedWhenLibOrRefFolderWithTFMDataIsNotFoundAndBuildFolderIsFound(string[] files)
+        {
+            files = files.ToArray();
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        [Theory]
+        [MemberData("WarningNotRaisedData", MemberType = typeof(FileSource))]
+        public void WarningNotRaisedWhenLibOrRefFolderWithTFMDataIsFound(string[] files)
+        {
+            files = files.ToArray();
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+    }
+
+    public static class FileSource
```

Comment:
Usually one class per file, but in cases like this the class should be nested inside the file's main class, rather than separate classes in a single file.